ols/src/auth/auth_dependency_interface.py

```diff
@@ -2,7 +2,33 @@
 
 from abc import ABC, abstractmethod
 
-from fastapi import Request
+from fastapi import HTTPException, Request
+
+
+def extract_bearer_token(header: str) -> str:
+    """Extract the bearer token from an HTTP authorization header.
+
+    Args:
+        header: The authorization header containing the token.
+
+    Returns:
+        The extracted token if present, else an empty string.
+    """
+    try:
+        scheme, token = header.split(" ", 1)
+        return token if scheme.lower() == "bearer" else ""
+    except ValueError:
+        return ""
+
+
+def extract_token_from_request(request: Request) -> str:
+    """Extract the bearer token from a FastAPI request."""
+    authorization_header = request.headers.get("Authorization")
+    if not authorization_header:
+        raise HTTPException(
```

Comment:
What else could cause this to fail? I am concerned we pass along `no user token` for other failures.

Reply:
It fail when there is no auth header. This is just moved logic from https://github.com/openshift/lightspeed-service/pull/2310/files#diff-c5c14c799ad92af71ed616b2187df94744b2986503a12ffad69f4a3fdc1e1ecdL266 to this function to be used in the noop - https://github.com/openshift/lightspeed-service/pull/2310/files#diff-3b24e1df8a702456e6bd68d43feab4a498270c7749577ab35a160cbf6b02fd36R40

Reply:
I know it fails when there is no auth header. I'm asking if anything else can make it fail, if so should we raise other exceptions that deal with that.

Reply:
Don't see the case.